homeassistant/components/mqtt/device_tracker/schema_discovery.py

```diff
@@ -121,46 +129,52 @@ def message_received(msg):
             },
         )
 
-    async def _subscribe_topics(self):
+    async def _subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
         await subscription.async_subscribe_topics(self.hass, self._sub_state)
 
     @property
-    def latitude(self):
+    def latitude(self) -> float | None:
         """Return latitude if provided in extra_state_attributes or None."""
         if (
             self.extra_state_attributes is not None
             and ATTR_LATITUDE in self.extra_state_attributes
         ):
-            return self.extra_state_attributes[ATTR_LATITUDE]
+            lat: float = self.extra_state_attributes[ATTR_LATITUDE]
+            return lat
         return None
 
     @property
-    def location_accuracy(self):
+    def location_accuracy(self) -> int:
         """Return location accuracy if provided in extra_state_attributes or None."""
         if (
             self.extra_state_attributes is not None
             and ATTR_GPS_ACCURACY in self.extra_state_attributes
         ):
-            return self.extra_state_attributes[ATTR_GPS_ACCURACY]
-        return None
+            accuracy: int = self.extra_state_attributes[ATTR_GPS_ACCURACY]
+            return accuracy
+        return 0
 
     @property
-    def longitude(self):
+    def longitude(self) -> float | None:
         """Return longitude if provided in extra_state_attributes or None."""
         if (
             self.extra_state_attributes is not None
             and ATTR_LONGITUDE in self.extra_state_attributes
         ):
-            return self.extra_state_attributes[ATTR_LONGITUDE]
+            long: float = self.extra_state_attributes[ATTR_LONGITUDE]
+            return long
         return None
 
     @property
-    def location_name(self):
+    def location_name(self) -> str | None:
         """Return a location name for the current location of the device."""
         return self._location_name
 
     @property
-    def source_type(self):
+    def source_type(self) -> SourceType | str | None:  # type: ignore[override]
```

Comment:
This should really be ignoring the type.
Either it is missing a validation step to enforce SourceType, or the base platform should be adjusted to allow strings.
cc @cdce8p

Reply:
Looking at the legacy device_tracker, it might be possible to allow `None`.
https://github.com/home-assistant/core/blob/147fe6f856a15ea4412866d3ce2d3e5fb7599cea/homeassistant/components/device_tracker/legacy.py#L704

Reply:
Solve by: https://github.com/home-assistant/core/pull/81128